src/tempo.py

```diff
@@ -201,39 +141,22 @@ def _build_storage_config(self, s3_config: dict):
                 queue_depth=20000,
             ),
             backend="s3",
-            s3=tempo_config.S3(
-                bucket=s3_config["bucket"],
-                access_key=s3_config["access-key"],
-                endpoint=s3_config["endpoint"],
-                secret_key=s3_config["secret-key"],
-            ),
+            s3=tempo_config.S3(**s3_config),
```

Comment:
beware of https://github.com/canonical/cos-lib/pull/36

Reply:
After checking, the `coordinator::_s3_config` is usable for Mimir, but not Tempo. In Tempo, S3 config has `secret_key` and `access_key` not `secret_access_key` and `access_key_id`

Reply:
haha lovely. Let's make a note then of why we're needing this transform step.